### ISSUE TITLE
anonymous types with vb.net

### ISSUE THREAD

**Issue by richardversluis** [serilog/serilog#618]
Hi,

Great library :). 

I have a question about using anonymous types in vb.net with serilog. When I do the following:

```vb.net
Dim someInfo = New With {.Bla = "bla", .Bla2 = "bla2"}
log.Information("{@SomeInfo}", someInfo)
```

The output is:

```
12/23/2015 11:48:10 VB$AnonymousType_1`2 { Bla: "bla", Bla2: "bla2" }
```

Like the example on the serilog.net homepage I wasn't expecting the type `VB$AnonymousType_1`2` to be logged, because it's anonymous. It this assumption wrong?

**Comment by nblumhardt** [comment]
Interesting - I wasn't aware VB has a different convention for naming anonymous types than C# does.

The check that makes the C# version work is at: https://github.com/serilog/serilog/blob/dev/src/Serilog/Parameters/PropertyValueConverter.cs#L158

There are various more comprehensive ways of checking, that would also work for VB, here: http://stackoverflow.com/questions/2483023/how-to-test-if-a-type-is-anonymous

This seems like a fix we can make, but needs some care to make sure things stay as quick as possible.

**Comment by nblumhardt** [comment]
This looks like a good implementation of the check: https://github.com/NancyFx/Nancy/blob/master/src/Nancy/ViewEngines/Extensions.cs#L29

**Comment by Pvlerick** [comment]
I'm happy to grab this if no-one is on it already.

**Comment by nblumhardt** [comment]
Closed in `dev` via #622.

**Comment by richardversluis** [comment]
:+1:
